### PR TITLE
Renew build.sh.

### PR DESCRIPTION
**Pull request by monaka**
### What does this PR do?

Adds two functions to build.sh.

* Dependency analysis among Dockfiles
* Pushing to the private repository.

### What issues does this PR fix or reference?

Nothing.

### New behavior

Resolves the build dependency by analyzing Dockerfiles .  
Enables to push the private repository by setting PRIVATE_REPO envvar.

### Tests written?
No. But I checked builds are done with no errors.

### Docs updated?
No. I couldn't find docs about build.sh script.

### CODE REVIEW

**Comment by eivantsov** [comment]
@monaka the script should assume that stack-base, php and python images are to be built before others.

File: build.sh [hunk 1]
```diff
@@ -1,43 +1,35 @@
...
+        docker push $PRIVATE_REPO/$image || error "Unable to push image: $image"
+    fi
+done
+
+for image in $eclipse_images; do
+    a=${image%:latest}
+    b=${a#eclipse/}
+    d=recipes/${b/:/\/}
+    echo $b $d
+    docker build -t $image $d || error "Unable to build image: $IMAGE_NAME"
```
Comment (eivantsov) [inline build.sh]: $IMAGE_NAME seems undefined here?

Comment (monaka) [inline build.sh]: Nice catch. Should be fixed here.

File: build.sh [hunk 2]
```diff
@@ -1,43 +1,35 @@
...
+    if [ "x$PRIVATE_REPO" != "x" ]; then
+        docker tag $image $PRIVATE_REPO/$image
+        docker push $PRIVATE_REPO/$image || error "Unable to push image: $IMAGE_NAME"
```
Comment (monaka) [inline build.sh]: Same here. ($IMAGE_NAME)

File: build.sh (hunk 1 above)
Comment (monaka) [inline build.sh]: Fixed now. Please review #107. Thanks for your support.

**Pull request merged**